pkg/cli/cvd.go

```diff
@@ -308,15 +318,65 @@ func (c *cvdCreator) createCVDFromLocalSrcs() ([]*hoapi.CVD, error) {
 	return res.CVDs, nil
 }
 
-func credentialsFactoryFromSource(source string) (CredentialsFactory, error) {
-	switch source {
+func credentialsFactoryFromSource(opts BuildAPICredentialsOpts) (CredentialsFactory, error) {
+	switch opts.Source {
 	case NoneCredentialsSource:
 		return func() string { return "" }, nil
 	case InjectedCredentialsSource:
 		return func() string { return client.InjectedCredentials }, nil
+	case FileCredentialSource:
+		return auth(opts.FilePath)
 	default:
-		return nil, fmt.Errorf("unknown credentials source: %s", source)
+		return nil, fmt.Errorf("unknown credentials source: %s", opts.Source)
+	}
+}
+
+type OAuthConfig interface {
+	auth() (*oauth2.Token, error)
+}
+
+type JwtConfig struct{ *jwt.Config }
+
+func NewJwtConfig(credential []byte) (*JwtConfig, error) {
+	jwtConfig, err := google.JWTConfigFromJSON(credential, "https://www.googleapis.com/auth/androidbuild.internal")
+	if err != nil {
+		return nil, err
+	}
+	return &JwtConfig{jwtConfig}, nil
+}
+
+func (c JwtConfig) auth() (*oauth2.Token, error) {
+	tk, err := c.TokenSource(context.Background()).Token()
+	if err != nil {
+		return nil, err
+	}
+	return tk, nil
+}
+
+// TODO: OAuth Config for end user
+
+func parseAuthConfig(credential []byte) (OAuthConfig, error) {
+	jwtConfig, err := NewJwtConfig(credential)
+	if err == nil {
+		return jwtConfig, nil
+	}
+	return nil, err
+}
+
+func auth(filepath string) (func() string, error) {
+	content, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read content from credential filepath %s: %w", filepath, err)
+	}
+	config, err := parseAuthConfig(content)
```

Comment:
Without using `parseAuthConfig`, perhaps we can change the code like below? Omitted error handling code in below example just for convenience.
```
config, err := NewJwtConfig(content)
tk, err := (OAuthConfig(config)).auth()
```